Expand GA analytics to include timing and exception tracking

humphd opened the pull request:
I notice that we can include timing and exception analytics in GA along with event categories.  I've expanded the API and added some new tracking code to help us see what's happening.

Fixes #2212

flukeout commented on public/editor/scripts/analytics.js:
```diff
@@ -65,14 +70,71 @@
+    if(options.label) {
+      eventOptions.timingLabel = trim(options.label);
+    }
+  }
```
The ``timing()`` function ends without ever calling ``gaSend``.

flukeout commented:
The documentation for exception shows the event structure as...

![image](https://cloud.githubusercontent.com/assets/25212/26474412/d3b1264c-4165-11e7-950a-23fd36415f83.png)

Yet when I console, log out the event object you're creating it's...

```
Object {
  hitType: "exception", 
  exDescription: "Unable to load Bramble editor in this browser", 
  exFatal: true
}
```

Will this alternate way work, or do we need to structure it like the docs say?

flukeout commented:
Also thanks for leaving links to the docs in the comments, that's a handy practice.

humphd commented:
OK, I've fixed the issues you found (you can indeed call the functions by passing the object vs. expanding the args out--both ways work).  I've also folded your timeout changes into this, and refactored our startup so we can cancel the timeout later. @gideonthomas you might want to take a look at this, too.

@flukeout can you test again, please?

flukeout commented:
Testing again now.

flukeout commented:
Just tested this and exceptions, timings and events seem to work well. The only difference I noticed is that you removed the ``eventLabel`` of the Black Screen troubleshooting event. We used to toString() the ``err`` and pass it along. Other than that looks good on my end.

flukeout commented:
cc @humphd

gideonthomas commented on public/editor/scripts/analytics.js:
```diff
@@ -65,14 +70,71 @@
+      if(window.performance) {
+        eventOptions["timingValue"] = Math.round(window.performance.now());
+      } else {
+        warn("Expected `value` arg to be a Number.");
```
wrong message?

gideonthomas commented:
This also fixes https://github.com/mozilla/thimble.mozilla.org/issues/2212
